### ISSUE TITLE
BOM generated incorrectly on new files when default encoding is utf8bom

### ISSUE THREAD

**Issue by diggityshmoo** [microsoft/vscode#102612]
Issue Type: <b>Bug</b>

Version: 1.47.0 (system setup)
Commit: d5e9aa0227e057a60c82568bf31c04730dc15dcd
Date: 2020-07-09T08:02:06.629Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.17763

When creating new ps1 files using this version of vscode it lists the default encoding as "UTF-8 with BOM", but i've been unable to source any of the files created as it returns an error of

The term '****' is not recognized as the name of a cmdlet, function, script file, or operable

where **** is whatever string appears first in the file.

Upon closer inspection i found that when copying the contents of the file to another editor there was an invisible character at the beginning of the file.  It looks like for UTF8-BOM files vscode is adding two copies of the BOM, and for files supposedly formatted as UTF8 with no BOM it's still putting the BOM once.

The file "test.ps1" was created by right clicking in the open files window and creating a new file (no contents were added to the file)  As you can see it contains two BOM:
```powershell
[12:03:49]:C:\Scripts\PowerShell\Scratch> (format-hex ".\test.ps1")[0]



           Path: C:\Scripts\PowerShell\Scratch\test.ps1

           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F

00000000   EF BB BF EF BB BF                                ï»¿ï»¿
```

after using the "save with encoding" option to save it as UTF-8 it now shows only one BOM (there should be nothing)

```powershell
[12:04:06]:C:\Scripts\PowerShell\Scratch> (format-hex ".\test.ps1")[0]



           Path: C:\Scripts\PowerShell\Scratch\test.ps1

           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F

00000000   EF BB BF                                         ï»¿
```

Creating a utf8 file with notepad++ results in an empty file, as expected:
```powershell
[12:09:54]:C:\Scripts\PowerShell\Scratch> (format-hex "..\ProfileScripts\notepadplusplus.ps1")[0]

Cannot index into a null array.
At line:1 char:1
+ (format-hex "..\ProfileScripts\notepadplusplus.ps1")[0]
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : NullArray
```

A UTF8-BOM file created in vscode on an earlier version also correctly shows only one BOM marker:

```powershell
[12:11:03]:C:\Scripts\PowerShell\Scratch> (format-hex ".\CheckSyncErrors.ps1")[0]



           Path: C:\Scripts\PowerShell\Scratch\CheckSyncErrors.ps1

           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F

00000000   EF BB BF 23 20 68 61 64 20 74 6F 20 6D 61 6E 75  ï»¿# had to manu
```

Something in this version appears to be causing it to encode these files incorrectly.

VS Code version: Code 1.47.0 (d5e9aa0227e057a60c82568bf31c04730dc15dcd, 2020-07-09T08:02:06.629Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8365U CPU @ 1.60GHz (8 x 1896)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.82GB (5.24GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
azure-pipelines|ms-|1.170.0
vscode-dotnet-runtime|ms-|0.1.2
python|ms-|2020.6.91350
azure-account|ms-|0.8.11
azurecli|ms-|0.5.0
powershell|ms-|2020.6.0
azurerm-vscode-tools|msa|0.11.0
html-preview-vscode|tht|0.2.5
shell-launcher|Tyr|0.4.1
vscode-icons|vsc|10.2.0

</details>

**Comment by diggityshmoo** [comment]
I just noticed that there was an update to 1.47.1 available, but i repeated the tests above and confirmed the problem is still present in that build

Version: 1.47.1 (system setup)
Commit: 485c41f9460bdb830c4da12c102daff275415b53
Date: 2020-07-13T22:42:53.386Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.17763

**Comment by diggityshmoo** [comment]
I did some additional testing and found that the problem only occurs when the default encoding is set to utf8bom.  When set to utf8 files are generated correctly with no bom and if they are converted to utf8bom they only have a single bom.

**Comment by gjsjohnmurray** [comment]
/duplicate of #102103 and will be fixed in upcoming 1.47.2
